accepted/00101-health-check-tool.md

```diff
@@ -0,0 +1,323 @@
+
+Goal: Set up Prometheus to scrape metrics from the Uyuni-Health-Exporter and evaluate alerting rules.
+
+Steps:
+
+  - Scrape Configuration: Add the Uyuni-Health-Exporter endpoint to the Prometheus configuration.
+  - Alerting Rules Definition: Create alerting rules to define conditions for triggering alerts based on metrics from the Uyuni-Health-Exporter.
+  - Alertmanager Configuration: Set up Alertmanager to handle alerts generated by Prometheus, including notifications.
+
+### Component 3: Grafana
+
+Goal: Use Grafana to visualize metrics and alerts from Prometheus.
+
+Steps:
+
+  - Add Prometheus as a data source in Grafana.
+  - Create dashboards to visualize metrics from the Uyuni-Health-Exporter.
+  - Configure panels within dashboards to display alerts based on Prometheus data.
+
+### Component 4: Loki and Promtail
+
+#### Loki setup
+
+Goal: Deploy and configure Loki to serve as the centralized log aggregation system for collecting, storing and querying logs from the Uyuni environment.
+
+Steps:
+
+  - Deployment: Install Loki on the Uyuni infrastructure as a container that is part of the Uyuni-Health-Check Pod or sharing the same network than the `uyuni-server` container.
+  - Configuration: Customize the Loki configuration to define storage locations for logs, retention policies and other operational parameters.
+  - Service Discovery: Configure Loki to discover targets for log collection, focusing on integration with the Uyuni-Health-Exporter and Uyuni components.
+
+#### Promtail setup
+
+Goal: Configure Promtail to collect logs from the Uyuni server and managed systems, forwarding them to Loki for aggregation and analysis.
+
+Steps:
+
+  - Deployment: Install Promtail on the Uyuni infrastructure as a container that is part of the Uyuni-Health-Check Pod.
+  - Configuration:
+    - Edit the Promtail configuration to define the paths of log files to monitor. In the containerized Uyuni environment, the Promtail container must be configured to have read access to mapped volumes that correspond to the log locations of the Uyuni server container.
+    - Configure Promtail to forward logs to the Loki instance.
+
+### Component 5: LogCLI
+Goal: Incorporate LogCLI to provide a command-line interface for querying logs stored in Loki, enhancing the toolset available for diagnostics, troubleshooting and operational reporting.
+
+
+### Component 6: Supportconfig metrics gatherer
+
+Stakeholders: Engineers and Supporters
+
+Goal: Extract relevant metrics from supportconfig files.
+
+Configuration: No extra configuration needed apart from the path to the supportconfig files.
+
+### Component 7: Saline
+
+Configuration: Deploy as a monitoring formula.
+
+
+## Alerting and notifications
+
+* Severity Levels: Alerts are categorized by severity levels (e.g., Critical, Warning, Info), allowing for a prioritized response and management.
+
+* Notification Channels: Configured within Grafana to support various notification mechanisms.
```

Comment:
As already commented above. We currently use _Alertmanager_ for configuring notifications.

Reply:
Yes, thank you. I have found one more appearance above and suggested a change there.